src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs

```diff
@@ -260,6 +250,10 @@ private void CloseWorker()
         /// </remarks>
         private PooledConnection GetPooledConnection(out SqlConnection connection)
         {
+            var scheduler = TaskScheduler.Current;
```

Comment:
We checked into this and there's no risk of a "surprise" where this doesn't do what we expected? If I had to be worried I'd mostly be worried that it wasn't pointing to the right scheduler and we might get some false exceptions but that's a perfectly safe way to fail.

Reply:
uh... that woudl be terrifying to me.  

@stephentoub TPL question.  We're using a ConcurrentExclusiveSchedulerPair to schedule work ot run on.  When that work runs, if that thread accesses TaskScheduler.Current it should see one of the scheduler from the schduler-pair right?

Reply:
> When that work runs, if that thread accesses TaskScheduler.Current it should see one of the scheduler from the schduler-pair right?

Yes.

Reply:
@stephentoub you da man :)

Reply:
Just to be entirely pedantic, the only way that wouldn't happen is if the person scheduling the task explicitly told it not to, e.g. when calling StartNew and passing in the target scheduler, if they also passed in TaskCreationOptions.HideScheduler.